src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -686,12 +686,6 @@ public static bool IsAutoPropertyAccessor(this IMethodSymbol methodSymbol)
         /// </summary>
         public static bool IsTopLevelStatementsEntryPointMethod([NotNullWhen(true)] this IMethodSymbol? methodSymbol)
             => methodSymbol?.ContainingType.IsTopLevelStatementsEntryPointType() == true &&
-               methodSymbol.IsStatic &&
-               methodSymbol.Name switch
-               {
-                   "$Main" => true,
-                   "<$Main>$" => true,
```

Comment:
We can just change this to `"<Main>$"`
https://github.com/dotnet/roslyn/blob/cf55f3a58e47298426fa971d3bd9d8857c746c65/src/Compilers/Core/Portable/Symbols/WellKnownMemberNames.cs#L351

Reply:
@mavasani When the code in tests is compiled, it's compiled against an older compiler version that produces `$Main` instead of `<Main>$`. I wanted to remove the old name from the extension method, so that required a newer version of the compiler. But as this is causing problems, I'll revert.

Reply:
Do you want to keep the issue open to track using the compiler API when it's good time to update the version?

Reply:
If you need new bits only for test you can bump up the below: https://github.com/dotnet/roslyn-analyzers/blob/193a3b623223f6c4a015d9faafb874fd739dff61/eng/Versions.props#L28

We should probably rename that property name to suggest that is for tests.

Reply:
> Do you want to keep the issue open to track using the compiler API when it's good time to update the version?

That sounds reasonable. Thanks.